### PR TITLE
update codebase to use badger in-memory database 

### PR DESCRIPTION
**Pull request by noot**
## Changes

- update codebase to use badger in-memory database instead of `chaindb.MemDatabase`
- update `TrieState` to use badger in-memory db (since it is transient anyways), clean up `StorageState.StoreTrie` accordingly

## Tests

```
go test ./... -short
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [ ] All integration tests and required coverage checks are passing

## Issues

- related to #1266

### CODE REVIEW

**Review by anthdm — APPROVED** [review]

File: dot/state/storage.go [hunk 1]
```diff
@@ -92,7 +92,11 @@ func (s *StorageState) pruneKey(keyHeader *types.Header) {
 
 // StoreTrie stores the given trie in the StorageState and writes it to the database
 func (s *StorageState) StoreTrie(root common.Hash, ts *rtstorage.TrieState) error {
-	var err error
+	// write raw trie key-values stored in db to *trie.Trie
+	err := ts.Commit()
+	if err != nil {
+		logger.Error("failed to write TrieState db values to trie", "state root", root, "error", err)
```
Comment (arijitAD) [inline dot/state/storage.go:98]: Any reason we aren't returning errors?

File: go.mod [hunk 2]
```diff
@@ -1,7 +1,7 @@
 module github.com/ChainSafe/gossamer
 
 require (
-	github.com/ChainSafe/chaindb v0.1.5-0.20210112170640-08c8f45c6f42
+	github.com/ChainSafe/chaindb v0.1.5-0.20210113205339-507b3a78c512
```
Comment (edwardmack) [inline go.mod:4]: Should we make a tag on chaindb repo for this so we can require a specific release (instead of master)?

**Review by edwardmack — APPROVED** [review]
Great work, lgtm.

File: dot/state/storage.go (hunk 1 above)
Comment (noot) [reply dot/state/storage.go:98]: it should be, will add!

File: go.mod (hunk 2 above)
Comment (noot) [reply go.mod:4]: hmm good idea, should I tag 0.1.0 at this commit?

Comment (edwardmack) [reply go.mod:4]: Not sure, what's the difference between releases and tags?  I see we're at release 0.0.1, but the tags are at v0.1.4.  Maybe 0.1.5 or 0.2.0, are these updates breaking changes?

**Pull request merged**